regression/cbmc/Pointer_comparison1/main.c

```diff
@@ -0,0 +1,22 @@
+#include <stdlib.h>
+
+int main()
+{
+  if(sizeof(void *) != 8)
+    return 0;
+
+  char *p = malloc(1);
+  if(p == 0)
+    return 0;
+
+  if(
+    (unsigned long)p >
+    42) // unsoundly evaluates to true due to pointer encoding
```

Comment:
Is this unsound?  ISO-9899 allows conversion of pointers to integers but doesn't say what they will be.  Thinking about actual platforms, this kind of thing could be ASLR dependent!

Reply:
Agreed, and I would thus argue that the assertion below ought to be considered reachable by a sound verification tool (i.e., there exists some system configuration under which `p` is less or equal to `42`).